lib/utils.c

```diff
@@ -90,6 +95,22 @@ int read_prop(char *dev, char *prop, long *value)
 	return -1;
 }
 
+/* Parse either a fraction '.3' or percent '30%
+ * return: 0 = ok, -1 = error, 1 = out of range
+ */
+int parse_percent(double *val, const char *str)
+{
+	char *p;
+
+	*val = strtod(str, &p) / 100.;
+	if (*val == HUGE_VALF || *val == HUGE_VALL)
+		return 1;
+	if (*p && strcmp(p, "%"))
```

Comment:
I think we should still have the case `*val == 0.0` here, because that's what is returned if for some reason no conversion is performed. Our usage of `sscanf()` before calling this function may make it so that a conversion can likely be performed, but not necessarily -- for example, what if "%%%" was passed into that `sscanf()` call. I think it would still pass, and then "%%%" would be passed to this function to try to convert to a double.

More generally, since this code is used by multiple files to parse a percentage, it should be prepared to handle all return values.

Reply:
Right, understood, will make the change.